nengo/spa/assoc_mem.py

```diff
@@ -68,17 +68,23 @@ def __init__(self, input_vocab, output_vocab=None,  # noqa: C901
             input_keys = input_vocab.keys
             input_vectors = input_vocab.vectors
         else:
-            input_vectors = input_vocab.create_subset(input_keys).vectors
+            # parse keys (allows expressions as keys)
+            input_vectors = [input_vocab.parse(key).v for key in
+                             input_keys]
 
         # If output vocabulary is not specified, use input vocabulary
         # (i.e autoassociative memory)
         if output_vocab is None:
             output_vocab = input_vocab
             output_vectors = input_vectors
+            output_keys = input_keys
```

Comment:
Yeah, looking at it now I think that line shouldn't be there. Unless @xchoo wants to share a bit more of his thoughts :)

 I added it because I thought keys would be used as class attributes/passed to the `network.AssociativeMemory`, but that doesn't seem to be the case. It seems like those keys are only used to create input and output vectors, if I'm not mistaking...